x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -67,7 +79,7 @@ regexBooleanExpression
     ;
 
 matchBooleanExpression
-    : qualifiedName MATCH_OPERATOR queryString=string
+    : valueExpression DEV_MATCH queryString=string
```

Comment:
@ioanatia try to stick with the existing grammar pattern; use valueExpression instead of qualifiedName. Perform validation if necessary in the parser .